packages/testing-library/lynx-environment/src/lynx/ElementPAPI.ts

```diff
@@ -0,0 +1,470 @@
+          // stop the propagation of the event
+          if (eventType === 'catchEvent' || eventType === 'capture-catch') {
+            evt.stopPropagation();
+          }
+          // @ts-ignore
+          globalThis.lynxCoreInject.tt.publishEvent(eventHandler, evt);
+        }
+      };
+      e.eventMap = e.eventMap ?? {};
+      e.eventMap[`${eventType}:${eventName}`] = listener;
+      e.addEventListener(
+        `${eventType}:${eventName}`,
+        listener,
+        {
+          // listening at capture stage
+          capture: eventType === 'capture-bind'
+            || eventType === 'capture-catch',
+        },
+      );
+    }
+
+    __GetEvent(e: LynxElement, eventType: string, eventName: string) {
+      const jsFunction = e.eventMap?.[`${eventType}:${eventName}`];
+      if (typeof jsFunction !== 'undefined') {
+        return {
+          type: eventType,
+          name: eventName,
+          jsFunction,
+        };
+      }
+      return undefined;
+    }
+
+    __SetID(e: LynxElement, id: string) {
+      e.id = id;
+    }
+
+    __SetInlineStyles(
+      e: LynxElement,
+      styles: string | Record<string, string>,
+    ) {
+      if (typeof styles === 'string') {
+        e.style.cssText = styles;
+      } else {
+        Object.assign(e.style, styles);
+      }
+    }
+
+    __AddDataset(e: LynxElement, key: string, value: string) {
+      e.dataset[key] = value;
+    }
+
+    __SetDataset(e: LynxElement, dataset: any) {
+      Object.assign(e.dataset, dataset);
+    }
+
+    __SetGestureDetector(
+      e: LynxElement,
+      id: number,
+      type: number,
+      config: any,
+      relationMap: Record<string, number[]>,
+    ) {
+      e.gesture = {
+        id,
+        type,
+        config,
+        relationMap,
+      };
+    }
+
+    __GetDataset(e: LynxElement) {
+      return e.dataset;
+    }
+
+    __RemoveElement(parent: LynxElement, child: LynxElement) {
+      let ch = parent.firstChild;
+      while (ch) {
+        if (ch === child) {
+          parent.removeChild(ch);
+          break;
+        }
+      }
+    }
+
+    __InsertElementBefore(
+      parent: LynxElement,
+      child: LynxElement,
+      ref?: LynxElement,
+    ) {
+      if (typeof ref === 'undefined') {
+        parent.appendChild(child);
+      } else {
+        parent.insertBefore(child, ref);
+      }
+    }
+
+    __ReplaceElement(
+      newElement: LynxElement,
+      oldElement: LynxElement,
+    ) {
+      const parent = oldElement.parentNode;
+      if (!parent) {
+        throw new Error('unreachable');
+      }
+      parent.replaceChild(newElement, oldElement);
+    }
+
+    __FlushElementTree(): void {}
+
+    __UpdateListComponents(_list: LynxElement, _components: string[]) {}
+
+    __UpdateListActions(
+      _list: LynxElement,
+      _removals: number[],
+      _insertions: number[],
+      _moveFrom: number[],
+      _moveTo: number[],
+      _updateFrom: number[],
+      _updateTo: number[],
+    ) {}
+
```

Comment:
I think this can be removed?

This `__UpdateListActions` is already removed in Lynx Engine.